File: src/Walrhouse.Application/UoMGroups/Queries/GetUoMGroups/UoMGroupDto.cs
using Walrhouse.Domain.Entities;
using Walrhouse.Domain.Enums;

namespace Walrhouse.Application.UoMGroups.Queries.GetUoMGroups;

public class UoMGroupDto
{
    public int Id { get; set; }
    public string BaseUoM { get; set; } = string.Empty;
    public List<UoMGroupLineDto> UoMGroupLines { get; set; } = new();

    public class Mapping : Profile
    {
        public Mapping()
        {
            CreateMap<UoMGroup, UoMGroupDto>();
            CreateMap<UoMGroupLine, UoMGroupLineDto>();
        }
    }
}

public class UoMGroupLineDto
{
    public string UoM { get; set; } = string.Empty;
    public int BaseQty { get; set; }
}

File: src/Walrhouse.Web/App/WebClient/src/features/receive/types/uomgroup-dto.ts
export interface UoMGroupLine {
  uoMName: string;
  baseQty: number;
}

export interface UoMGroupDto {
  id: number;
  baseUoMName: string;
  uoMGroupLines: UoMGroupLine[];
}

export interface CreateUoMGroupDto {
  baseUoMName: string;
  uoMGroupLines: UoMGroupLine[];
}

export interface UpdateUoMGroupDto {
  id: number;
  baseUoMName: string;
  uoMGroupLines: UoMGroupLine[];
}
